src/test/java/specs/TimeoutSpecs.java

```diff
@@ -0,0 +1,111 @@
+        final Result result = SpectrumHelper.run(() -> {
+          describe("Suite with low timeout", with(timeout(1), () -> {
+            it("has spec oversleeps and fails", () -> {
+              Thread.sleep(1000);
+            });
+          }));
+        });
+        assertThat(result.getFailureCount(), is(1));
+      });
+
+      it("will report a timing out test correctly when there are surrounding hooks", () -> {
+        // if hooks were run in within the timeout's envelope, then when the child
+        // is killed for timing out, there would be "bleed" of reporting
+        final Result result = SpectrumHelper.run(() -> {
+          describe("Suite with low timeout", with(timeout(1), () -> {
+            Supplier<String> let = let(() -> "Hello world");
+            beforeEach(() -> {
+              // deliberately blank before each - just for ensuring a hook is present
+            });
+            it("has spec oversleeps and fails", () -> {
+              assertThat(let.get(), is("Hello world"));
+              Thread.sleep(1000);
+            });
+          }));
+        });
+        assertThat(result.getFailureCount(), is(1));
+      });
+
+      it("will fail the test that fails its timeout and pass the others", () -> {
+        final Result result = SpectrumHelper.run(() -> {
+          describe("Suite with timeout", with(timeout(10), () -> {
+            it("has spec that takes no time", () -> {
+
+            });
+
+            it("has spec that oversleeps and fails", () -> {
+              Thread.sleep(1000);
+            });
+
+            it("has another spec that takes no time", () -> {
+
+            });
+          }));
+        });
+        assertThat(result.getRunCount(), is(3));
+        assertThat(result.getFailures().get(0).getDescription().getMethodName(),
+            is("has spec that oversleeps and fails"));
+        assertThat(result.getFailureCount(), is(1));
+      });
+
+      it("will propagate the timeout from parent to child", () -> {
+        final Result result = SpectrumHelper.run(() -> {
+          describe("Suite with low timeout", with(timeout(1), () -> {
+            describe("with child suite", () -> {
+              it("oversleeps and fails", () -> {
+                Thread.sleep(1000);
+              });
+            });
+          }));
+        });
+        assertThat(result.getFailureCount(), is(1));
+      });
+
+      it("will allow a lower level to supersede the parent's timeout", () -> {
+        final Result result = SpectrumHelper.run(() -> {
+          describe("Suite with low timeout", with(timeout(1, TimeUnit.MILLISECONDS), () -> {
+            describe("with child suite", () -> {
+              it("has a more generous timeout", with(timeout(1, TimeUnit.MINUTES), () -> {
+                Thread.sleep(100);
+              }));
+            });
+          }));
+        });
+        assertThat(result.getFailureCount(), is(0));
```

Comment:
I noticed all the assertions here are confirming that tests fail/pass in the right scenarios. What about the messaging and user feedback that goes along with that? I haven't reviewed the implementation yet, but I assume there are some helpful messages that show up when one of these fail ("Timeout expired" or something to that effect).

Reply:
Ah, I see it's leveraging JUnit4's [`FailOnTimeout` helper](https://github.com/junit-team/junit4/blob/master/src/main/java/org/junit/internal/runners/statements/FailOnTimeout.java). Might still be worth verifying some of that behavior in case we end up switching to something else down the line (JUnit5 might have a different helper, or we write our own).

Reply:
We could wrap the helper's exception with a Spectrum one and then verify that, giving a seamless switch between implementations, or we could not promise anything special beyond _it fails_. What do you think?

Reply:
I'm inclined not to worry about the exact failure - only that it fails.